crates/sui-surfer/src/surfer_task.rs

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{collections::HashMap, sync::Arc};
+
+use rand::{rngs::StdRng, Rng, SeedableRng};
+use sui_core::authority::authority_store_tables::LiveObject;
+use sui_types::{
+    base_types::{ObjectRef, SuiAddress},
+    object::Owner,
+};
+use test_utils::network::TestCluster;
+use tokio::sync::{watch, RwLock};
+
+use crate::{
+    surf_strategy::SurfStrategy,
+    surfer_state::{ImmObjects, OwnedObjects, SharedObjects, SurfStatistics, SurferState},
+};
+
+pub struct SurferTask {
+    pub state: SurferState,
+    pub surf_strategy: Box<dyn SurfStrategy>,
+    pub exit_rcv: watch::Receiver<()>,
+}
+
+impl SurferTask {
+    pub async fn create_surfer_tasks<S: Default + SurfStrategy>(
+        cluster: Arc<TestCluster>,
+        rng: &mut StdRng,
+        exit_rcv: watch::Receiver<()>,
+        immutable_objects: ImmObjects,
+        shared_objects: SharedObjects,
```

Comment:
`create_surfer_tasks` is only called during initialization time with both `immutable_objects` and `shared_objects` being empty hashmaps. Why not initialize `immutable_objects` and `shared_objects` inside this function then?